src/h5web/App.test.tsx

```diff
@@ -126,9 +126,115 @@ test('navigate groups in explorer', async () => {
 
   expect(queryByRole('treeitem', { name: '1_cormap' })).not.toBeInTheDocument();
 
-  // `waitFor` avoids warning due to async work occuring after completion of test
+  // `waitFor` avoids warning due to async work occurring after completion of test
   await waitFor(() => {
     expect(groupBtn).toHaveAttribute('aria-selected', 'true');
     expect(groupBtn).toHaveAttribute('aria-expanded', 'false');
   });
 });
+
+test('select a scalar dataset and display the value', async () => {
+  const { findByRole, getByRole, getByText } = renderApp();
+
+  // Select the scalar dataset entry_0000/1_cormap/version
+  fireEvent.click(await findByRole('treeitem', { name: 'entry_0000' }));
+  fireEvent.click(getByRole('treeitem', { name: '1_cormap' }));
+  const datasetBtn = getByRole('treeitem', { name: 'version' });
+  expect(datasetBtn).toBeVisible();
+  fireEvent.click(datasetBtn);
+  expect(datasetBtn).toHaveAttribute('aria-selected', 'true');
+  expect(datasetBtn).toHaveAttribute('aria-expanded', 'false'); // A dataset cannot be expanded
+
+  const scalarTab = await findByRole('tab', { name: 'Scalar' });
+  expect(scalarTab).toBeVisible();
+  expect(scalarTab).toHaveAttribute('aria-selected', 'true');
+
+  const datasetValue = getByText('0.7.1');
+  expect(datasetValue).toBeVisible();
+});
+
+test('switch visualisations for a 1D dataset', async () => {
+  const { findByRole, getByRole } = renderApp();
+
+  // Select the 1D dataset entry_0000/1_cormap/to_merge
+  fireEvent.click(await findByRole('treeitem', { name: 'entry_0000' }));
+  fireEvent.click(getByRole('treeitem', { name: '1_cormap' }));
+  const datasetBtn = getByRole('treeitem', { name: 'to_merge' });
+  expect(datasetBtn).toBeVisible();
+  fireEvent.click(datasetBtn);
+
+  const visSelector = await findByRole('tablist', { name: 'Visualization' });
+  expect(visSelector).toBeVisible();
+
+  const lineTab = getByRole('tab', { name: 'Line' });
+  expect(lineTab).toBeVisible();
+  expect(lineTab).toHaveAttribute('aria-selected', 'true');
+
+  const matrixTab = getByRole('tab', { name: 'Matrix' });
+  expect(matrixTab).toBeVisible();
+  expect(matrixTab).toHaveAttribute('aria-selected', 'false');
+
+  // Switching visualisations
+  fireEvent.click(matrixTab);
+  expect(matrixTab).toHaveAttribute('aria-selected', 'true');
+  expect(lineTab).toHaveAttribute('aria-selected', 'false');
+});
+
+test('switch visualisations for a 2D dataset', async () => {
```

Comment:
Not sure this one is very valuable since you've already tested that the vis selector works and we have utility tests that check that supported visualizations are computed correctly. How about testing the matrix vis, perhaps? You could check the number of rows/cols for instance.

Then, the previous test could be renamed simply `switch visualization`, and you could add another test to check the matrix vis with a 1D dataset.